base/cvd/cuttlefish/host/commands/cvd/instances/instance_record.cpp

```diff
@@ -99,6 +100,31 @@ Result<Json::Value> LocalInstance::FetchStatus(std::chrono::seconds timeout) {
 }
 
 Result<void> LocalInstance::PressPowerBtn() {
+  auto bin_check = 
+      HostToolTarget(host_artifacts_path()).GetPowerBtnBinPath();
+  if (bin_check.ok()) {
+    return PressPowerBtnLegacy();
+  }
+
+  std::unique_ptr<const CuttlefishConfig> config = CuttlefishConfig::GetFromFile(instance_dir() + "/cuttlefish_config.json");
+  auto instance = config->ForInstance(id());
+
+  Command command(instance.crosvm_binary());
```

Comment:
This should verify that vm manager is indeed crosvm before attempting a crosvm command.

Reply:
Done.